### PR TITLE
Add missing GenericScript extension

### PR DESCRIPTION
**Pull request by pl-buiquang**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
Generic script afs object does not have an associated extension

**What is the new behavior (if this is a feature change)?**
The extension now allows the creation of generic script objects

### CODE REVIEW

**Comment by jonenst** [comment]
Why is the AfsCircularDependencyException.class test removed ?

**Comment by jonenst** [comment]
It feels like adding the invalidate() calls in the add and remove methods could have been in a separate PR (or is it really tied to adding the extension and the builder ?)
Same for adding a script type ?
same for setting the aspectj scope

**Comment by pl-buiquang** [comment]
> Why is the AfsCircularDependencyException.class test removed ?

It was not removed, just moved (I guess, or is it an other one ?)

**Comment by pl-buiquang** [comment]
> It feels like adding the invalidate() calls in the add and remove methods could have been in a separate PR (or is it really tied to adding the extension and the builder ?)
> Same for adding a script type ?
> same for setting the aspectj scope

I agree that is PR contains fixes not entirely related to the addition of GenericScript extension. Yet it does belong to the broader issue relating to virtual scripts and/or are simplistic fixes that would waste my time creating a single PR for..

**Comment by jonenst** [comment]
> It was not removed, just moved (I guess, or is it an other one ?)

I'm talking about the 5 lines around line 154 in ModificationScriptTest.java
```
     assertThatCode(() -> genericScript1.addGenericScript(genericScript1)).isInstanceOf(AfsCircularDependencyException.class);
```

File: afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ModificationScriptTest.java [hunk 1]
```diff
@@ -151,14 +161,9 @@ public void test() {
...
 
-        NodeInfo genScript1NodeInfo = storage.createNode(rootFolder.getId(), "genScript1", GenericScript.PSEUDO_CLASS, "", GenericScript.VERSION, new NodeGenericMetadata());
-        ProjectFileCreationContext projectFileCreationContext = new ProjectFileCreationContext(genScript1NodeInfo, storage, project);
-        GenericScript genericScript1 = new GenericScript(projectFileCreationContext);
-        storage.setConsistent(genScript1NodeInfo.getId());
-        genericScript1.writeScript("list of things");
-        storage.flush();
-        script.addGenericScript(genericScript1);
-
-        assertThatCode(() -> genericScript1.addGenericScript(genericScript1)).isInstanceOf(AfsCircularDependencyException.class);
```
Comment (jonenst) [inline afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ModificationScriptTest.java]: why is this test removed?

Comment (pl-buiquang) [reply afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ModificationScriptTest.java]: it was moved up

Comment (pl-buiquang) [reply afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ModificationScriptTest.java]: line 153

Comment (jonenst) [reply afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ModificationScriptTest.java]: You're right, I missed it.

**Pull request merged**